### ISSUE TITLE
path: compile error on Fedora 33 ASAN GCC 10.2.1

### ISSUE THREAD

**Issue by mpranj** [ElektraInitiative/libelektra#3552]
## Steps to Reproduce the Problem

Build Elektra on on Fedora 33 with ASAN using GCC 10.2.1.

## Actual Result

```
/home/jenkins/workspace/libelektra_PR-3532/src/plugins/path/path.c: In function 'validateKey':
/home/jenkins/workspace/libelektra_PR-3532/src/plugins/path/path.c:72:7: error: argument 2 is null but the corresponding size argument 3 value is 1000 [-Werror=nonnull]
   72 |   if (strerror_r (errno, errmsg, ERRORMSG_LENGTH) != 0)
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/jenkins/workspace/libelektra_PR-3532/src/include/kdbplugin.h:16,
                 from /home/jenkins/workspace/libelektra_PR-3532/src/plugins/path/path.h:15,
                 from /home/jenkins/workspace/libelektra_PR-3532/src/plugins/path/path.c:9:
/usr/include/string.h:428:14: note: in a call to function 'strerror_r' declared with attribute 'write_only (2, 3)'
  428 | extern char *strerror_r (int __errnum, char *__buf, size_t __buflen)
      |              ^~~~~~~~~~
```

Error discovered in #3532.

## System Information

- Elektra Version: master

## Further Log Files and Output
[log_path.txt](https://github.com/ElektraInitiative/libelektra/files/5504505/log_path.txt)
[Jenkins CI run](https://build.libelektra.org/blue/organizations/jenkins/libelektra/detail/PR-3532/4/pipeline/678)

**Comment by markus2330** [comment]
Thank you for reporting this!

Good that we did not fix #2794 yet, otherwise we would have this warning everywhere :smile: 

I still think we should consider a generic solution to somehow embedding strerror_r into ELEKTRA_ADD_RESOURCE_WARNINGF, as it is basically always needed and the "easy" way using strerror is always a bug (#2794) and this strcat approach is terrible...

**Comment by mpranj** [comment]
The error was actually simple: the malloc return value was not checked. I'm just surprised that this error does not pop up at more locations.
